src/explorer/ColumnExploration.cs

```diff
@@ -2,37 +2,60 @@ namespace Explorer
 {
     using System;
     using System.Collections.Generic;
+    using System.Linq;
     using System.Threading.Tasks;
 
+    using Diffix;
     using Explorer.Metrics;
-    using Lamar;
+    using Microsoft.Extensions.Logging;
 
-    public sealed class ColumnExploration : IDisposable
+    public sealed class ColumnExploration : AbstractExploration, IDisposable
     {
-        private readonly MetricsPublisher publisher;
-        private readonly INestedContainer scope;
+        private readonly ExplorationScope scope;
         private bool disposedValue;
 
-        public ColumnExploration(ExplorationConfig config, INestedContainer scope, string column)
+        public ColumnExploration(ExplorationScope scope)
         {
+            if (scope.Ctx is null)
+            {
+                throw new InvalidOperationException(
+                    $"Can't build {GetType().Name} without a context object in scope!");
+            }
+
             this.scope = scope;
-            publisher = scope.GetInstance<MetricsPublisher>();
-            Column = column;
-            Completion = Task.WhenAll(config.Tasks);
         }
 
-        public string Column { get; }
+        public string Column { get => Ctx.Column; }
 
-        public Task Completion { get; }
+        public DColumnInfo ColumnInfo { get => Ctx.ColumnInfo; }
 
-        public IEnumerable<ExploreMetric> PublishedMetrics => publisher.PublishedMetrics;
+        public ExplorerContext Ctx { get => scope.Ctx!; }
```

Comment:
It is possible to remove the null forgiving operator from here if you make the type of this `ExplorationScope.Ctx` not-nullable, and move the throw from the `ColumnExploration` constructor in the `ExplorationScope.Ctx`.

Reply:
How about this
26b868e8a4705587393dba2ff80045ba6e63fd22